source/lac/cuda_sparse_matrix.cu

```diff
@@ -27,6 +27,28 @@ DEAL_II_NAMESPACE_OPEN
 
 namespace CUDAWrappers
 {
+  namespace
+  {
+    template <typename Number>
+    void
+    delete_device_vector(Number *device_ptr) noexcept
```

Comment:
It's not really a vector here. Maybe `delete_device_data()`? Same for `allocate...`?

Reply:
I happily following your naming suggestion. Once #7075 and #7084 are merged, I suggest to unify all these functions in `base/cuda.h` anyway.